pkg/agents/team4/finalAgent/agent.go

```diff
@@ -0,0 +1,197 @@
+	ageLastEaten       int              // the age at which the agent last ate
+	morality           float64          // the morality of the agent that determines how selfishly or selflessly the agent will act
+	traumaScaleFactor  float64          // the amount of trauma the agent has suffered which effects the amount of food it is likely to eat
+
+	globalTrust      float32            // the overall trust the agent has in other agents in the tower
+	coefficients     []float32          // the amount trust score changes by for certain actions
+	lastFoodTaken    food.FoodType      // food taken on the previous day
+	sentMessages     []messages.Message // TODO: make it a map hashed by messageIDs
+	responseMessages []messages.Message // TODO: make it a map hashed by messageIDs
+	lastPlatFood     food.FoodType      // last seen food on the platform
+	maxFoodLimit     food.FoodType      // maximum food we want to allow others to eat
+	messageCounter   int                // the total number of messages we send in a day
+	globalTrustLimit float32            // limit to check whether to be selfish or not
+	lastAge          int                // the age of the agent on the previous day
+	healthStatus     int
+}
+
+type CustomAgentEvo struct {
+	*infra.Base
+	// new params
+	params CustomAgentEvoParams
+}
+
+// Data struct to hold evo param values
+type LoadedData struct {
+	FoodToEat  []int
+	DaysToWait []int
+}
+
+func InitaliseParams(baseAgent *infra.Base) CustomAgentEvoParams {
+
+	foodToEat := map[string][]int{
+		"selfish":  {baseAgent.HealthInfo().HPReqCToW, 7, 9, 11}, // TODO: to optimise more baseAgent.HealthInfo().HPReqCToW
+		"selfless": {baseAgent.HealthInfo().HPReqCToW, 7, 9, 11}, // TODO: to optimise more
+	}
+	daysToWait := map[string][]int{
+		"selfish":  {0, 1, 4, 3}, // TODO: to optimise more int(baseAgent.HealthInfo().MaxDayCritical / 2)
+		"selfless": {0, 1, 4, 3}, // TODO: to optimise more
+	}
+
+	return CustomAgentEvoParams{ //initialise the parameters of the agent
+		locked:             true,
+		foodToEat:          foodToEat,
+		daysToWait:         daysToWait,
+		currentPersonality: "selfless",
+		ageLastEaten:       0,
+		morality:           100 * rand.Float64(), // TODO: Use this properly
+		traumaScaleFactor:  1,
+		healthStatus:       3,
+		globalTrust:        0.0,
+		coefficients:       []float32{2, 4, 8}, // TODO: maybe train these co-efficients using evolutionary algorithm
+		lastFoodTaken:      0,
+		sentMessages:       []messages.Message{},
+		responseMessages:   []messages.Message{},
+		lastPlatFood:       -1,
+		maxFoodLimit:       50,
+		messageCounter:     0,
+		globalTrustLimit:   75,
+		lastAge:            0,
+	}
+}
+
+func New(baseAgent *infra.Base) (infra.Agent, error) {
+	//create other parameters
+	return &CustomAgentEvo{
+		Base:   baseAgent,
+		params: InitaliseParams(baseAgent),
+	}, nil
+}
+
+// Checks if neighbour below has eaten
+func (a *CustomAgentEvo) NeighbourFoodEaten() food.FoodType {
```

Comment:
Most (if not all) of your functions should not be exported, but it's a minor thing - you can fix it in your next PR if you need this one out asap.